programs/bpf_loader/src/lib.rs

```diff
@@ -470,6 +470,10 @@ fn process_loader_upgradeable_instruction(
                 log!(logger, "Program account not executable");
                 return Err(InstructionError::AccountNotExecutable);
             }
+            if !program.is_writable() {
```

Comment:
We'll need to gate this.  Otherwise there's a window between 33%-66% of stake upgrade where you can trivially break consensus by upgrading a program with an older cli